### PR TITLE
Synchronization 

### PR DESCRIPTION
**Pull request by bblaszkow06**
Closely related to https://github.com/membraneframework/membrane-core/pull/160

### CODE REVIEW

File: guide/creating_app/advanced.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# More advanced pipelines
+
+After succcessfully running your first Membrane Pipeline, we can move on to more complicated stuff.
+In this guide, we will show you how to provide options for pads, how to link an element with dynamic pad(s)
+and how introduce synchronization of start between elements
```
Comment (kanes115) [inline guide/creating_app/advanced.md]: how to

File: guide/creating_app/advanced.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+## Pad options
+
+Not only elements can have some options. Some pads have options as well.
```
Comment (kanes115) [inline guide/creating_app/advanced.md]: Not only elements can have ~some~ options

File: guide/creating_app/advanced.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+You can provide them in a keyword at the end of the tuple defining link:
```
Comment (kanes115) [inline guide/creating_app/advanced.md]: Keyword list?

File: guide/creating_element/synchronization.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+* (optionally) pid of a clock that should be used by the timer. If no clock provided, a pipeline clock is used.
+
+Starting a clock means that the element will start receiving ticks that should be handled by a new callback -
+`c:Membrane.Element.Base.handle_tick/3` receiving timer id along with context and state.
+
+The timer can be stopped by `t:Membrane.Element.Action.stop_timer_t/0` action with id of a timer.
+
+### Example
+
+Here's a simplified example of a sink element that needs to consume video frames in proper speed and uses timer for that.
```
Comment (Hajto) [inline guide/creating_element/synchronization.md]: How about: `Here's a simplified example of a sink element that uses timer to consume video frames in proper speed.`

**Pull request merged**